ui-components/src/actions/FavoriteButton.tsx

```diff
@@ -0,0 +1,128 @@
+    sendSmsNotifications: false,
+    favoriteIds: [],
+  }
+
+  const [listingFavorited, setListingFavorited] = useState(preferences.favoriteIds?.includes(id))
+
+  useEffect(() => {
+    setListingFavorited(preferences.favoriteIds?.includes(id))
+  }, [preferences.favoriteIds, id])
+
+  if (!profile) {
+    return <span />
+  }
+
+  const addFavorite = async () => {
+    if (!profile || listingFavorited) {
+      return
+    }
+    preferences.favoriteIds?.push(id)
+
+    try {
+      await userPreferencesService?.update({
+        id: profile.id,
+        body: preferences,
+      })
+      setListingFavorited(true)
+    } catch (err) {
+      console.warn(err)
+    }
+  }
+
+  const removeFavorite = async () => {
+    if (!profile || !preferences.favoriteIds || preferences?.favoriteIds?.length === 0) {
+      return
+    }
+
+    const index: number = preferences.favoriteIds?.indexOf(id)
+    preferences?.favoriteIds?.splice(index, 1)
+
+    try {
+      await userPreferencesService?.update({
+        id: profile.id,
+        body: preferences,
+      })
+      setListingFavorited(false)
+    } catch (err) {
+      console.warn(err)
+    }
+  }
+
+  return (
+    <span>
+      {listingFavorited ? (
+        <span>
+          <Button
+            className="mx-2 p-3 rounded-full bg-primary-dark border-primary-dark"
+            size={AppearanceSizeType.small}
+            onClick={() => removeFavorite()}
+            ariaLabel={t("t.unfavorite")}
+          >
+            <Icon
+              symbol={"likeFill"}
+              size={"extra-medium"}
+              fill={IconFillColors.white}
+              iconClass={"favorited-fill mt-0"}
+            />
+          </Button>
+          <a
+            role="button"
+            aria-label={t("t.unfavorite")}
+            onClick={removeFavorite}
+            className={"cursor-pointer font-bold align-middle"}
+          >
+            {t("t.favorite")}
+          </a>
+        </span>
+      ) : (
+        <span>
+          <Button
+            className="mx-2 p-3 rounded-full"
+            size={AppearanceSizeType.small}
+            onClick={() => addFavorite()}
+            ariaLabel={t("t.favorite")}
+          >
+            <Icon symbol={"like"} size={"extra-medium"} iconClass={"mt-0"} />
+          </Button>
+          <a
+            role="button"
+            aria-label={t("t.favorite")}
+            onClick={addFavorite}
+            className={"cursor-pointer font-bold align-middle"}
+          >
+            {t("t.favorite")}
+          </a>
+        </span>
+      )}
+      {/* Below uses aria-live which announces changes to a screen reader. The element must be "visible" to the screen reader, hence the classes on there. */}
+      <span
+        className="text-white text-xs max-h-px max-w-0 inline-block overflow-hidden"
+        aria-live="assertive"
+      >
+        {listingFavorited ? `${t("t.favorite")} ${name}` : `${t("t.unfavorite")} ${name}`}
```

Comment:
I think the verbiage seems a little off here. I feel like it should be `Favorited <listing name>` or `Unfavorited <listing name>` on first pass I thought this was saying you can now favorite the listing

Reply:
@YazeedLoonat , I updated to be past tense.